### ISSUE TITLE
SDL_BLENDOPERATION_SUBTRACT documentation is incorrect.

### ISSUE THREAD

**Issue by AliceLR** [libsdl-org/SDL#8910]
Both the [SDL2](https://wiki.libsdl.org/SDL2/SDL_BlendOperation) and [SDL3](https://wiki.libsdl.org/SDL3/SDL_BlendOperation) documentation (as well as the [doc comments in SDL_blendmode.h](https://github.com/libsdl-org/SDL/blob/6fb188eb35f84eeb0b77c351f83730478072eb99/include/SDL3/SDL_blendmode.h#L68)) currently claim that `SDL_BLENDOPERATION_SUBTRACT` is `dst - src` and that `SDL_BLENDOPERATION_REV_SUBTRACT` is `src - dst`.

This doesn't seem to match SDL's actual behavior or any of the underlying graphics APIs, where `GL_FUNC_SUBTRACT` et al. are `src - dst` and `GL_FUNC_REVERSE_SUBTRACT` et al. are `dst - src`: [OpenGL](https://docs.gl/gl4/glBlendEquation), [GLES](https://docs.gl/es2/glBlendEquation), [D3D](https://learn.microsoft.com/en-us/windows/win32/direct3d9/d3dblendop), [Metal](https://developer.apple.com/documentation/metal/mtlblendoperation), and also [Vulkan](https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/VkBlendOp.html) for good measure.

Code fragment: this blendmode/color combo inverts the destination color and sets the alpha to 1.0 (SDL 2.28.5, Windows 10, GeForce GTX 660, `opengl` and `direct3d` render drivers tested), which wouldn't work if the `SDL_BLENDOPERATION_SUBTRACT` documentation was correct:
```C
    SDL_BlendMode invert = SDL_ComposeCustomBlendMode(
      SDL_BLENDFACTOR_ONE, SDL_BLENDFACTOR_ONE, SDL_BLENDOPERATION_SUBTRACT,
      SDL_BLENDFACTOR_ONE, SDL_BLENDFACTOR_ZERO, SDL_BLENDOPERATION_SUBTRACT);

    SDL_SetRenderDrawBlendMode(renderer, invert);
    SDL_SetRenderDrawColor(renderer, 255, 255, 255, 255);
    SDL_RenderFillRect(renderer, &dest);
```
(I think the docs also don't mention that for OpenGL both operations provided to `SDL_ComposeCustomBlendMode` [need to be the same](https://github.com/libsdl-org/SDL/blob/6fb188eb35f84eeb0b77c351f83730478072eb99/src/render/opengl/SDL_render_gl.c#L400), but I don't know if that matters.)

**Comment by slouken** [comment]
@icculus, the documentation is correct here, right?

**Comment by icculus** [comment]
Sure seems like it isn't!

I'll look later today, but this seems like a correct bug report.

**Comment by slouken** [comment]
I think we can flip these for SDL3 and swap them in sdl2-compat.

**Comment by icculus** [comment]
> I think we can flip these for SDL3 and swap them in sdl2-compat.

It's just the docs that are wrong, though, they've always behaved the way OpenGL, etc, expects them to, so flipping what they currently do would actually be incorrect.

**Comment by slouken** [comment]
Oh, sorry, I misunderstood.

**Comment by icculus** [comment]
Okay, docs are corrected in SDL3, SDL2, and release-2.30.x branches, and the SDL2 and SDL3 wikis!
